### PR TITLE
Add validations and model specs

### PR DESCRIPTION
**Pull request by NatiDeme**
# Changes in this PR:
- No linter errors
- Included Validations:
   **For the User model:**
    - Name must not be blank.
   - PostsCounter must be an integer greater than or equal to zero.

    **For the Post model:**

   -  Title must not be blank.
   -  Title must not exceed 250 characters.
   -  CommentsCounter must be an integer greater than or equal to zero.
   -  LikesCounter must be an integer greater than or equal to zero.
  
- Imlemented unit specs for all of your models' methods and validations.

### CODE REVIEW

File: app/models/specs/models/like_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  it 'tests that Like model is created correctly' do
+    like = Like.new
+    like.build_user(name: 'Lolade')
+    like.build_post(text: 'new post 2')
+    expect(like).to be_valid
+  end
```
Comment (dasileker) [inline app/models/specs/models/like_spec.rb]: - [x] Hello and excellent work, but the spec files should be created using the Rpsec.

File: app/models/specs/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      expect(subject).to_not be_valid
+    end
+
+    it 'Title should not be greater than 250 characters' do
+      expect(subject).to be_valid
+    end
+
+    it 'Comments counter should be integer' do
+      subject.comments_counter = 2
+      expect(subject).to be_valid
```
Comment (dasileker) [inline app/models/specs/models/post_spec.rb]: - [x] The same thing is also should e done for this file.

File: app/models/specs/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    before { subject.save }
+
+    it 'Check name should be present' do
+      subject.name = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'User should have post greater than or equal to 0' do
+      subject.posts_counter = -1
+      expect(subject).to_not be_valid
```
Comment (dasileker) [inline app/models/specs/models/user_spec.rb]: - [x] also this file

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @NatiDeme,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**
- Linters are green ✅.
- PR is very nice 👌.

## Required Changes ♻️
_Check the comments under the review._

- Hello and good job on the validation and spec but you have added the spec files incorrectly. the spec files should be added using the RSpec method, [check this tutorial about rspec](https://www.tutorialspoint.com/rspec/rspec_introduction.htm)
- Also please make sure that the readme file is having all the necessary elements in the **Getting started section** like how to install the gem and run the migration.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
+require 'rails_helper' # Return character is CR+LF on Windows.
+
+RSpec.describe Post, type: :model do
+  describe 'Post validations' do
+    user1 = User.create(name: 'Lolade', bio: 'Public Administator', posts_counter: 0)
```
Comment (GzimAsani) [inline spec/models/post_spec.rb]: - [x] Unfortunately, now I am getting this error when trying to run your tests :confused: 
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @NatiDeme              :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your project  :+1:  
 
 ## Required Changes 

- [x] Please implement the changes on your readme file asked by the previous reviewer and don't forget to include in your readme file the steps on how to test your project
- [x] Check the comments below :arrow_double_down: 






 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rustam-tolipov — APPROVED** [review]
## Hello @NatiDeme 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**